tests/unit/test_base_collection.py

```diff
@@ -0,0 +1,133 @@
+from unittest import TestCase
+
+import responses
+
+from tamr_unify_client import Client
+from tamr_unify_client.auth import UsernamePasswordAuth
+
+
+class DeleteByResourceID(TestCase):
+    def setUp(self):
+        auth = UsernamePasswordAuth("username", "password")
+        self.unify = Client(auth)
+
+    @responses.activate
+    def test_dataset_collection(self):
+        ds_url = self.url_prefix + "datasets/115"
+
+        responses.add(responses.DELETE, ds_url, status=204)
+
+        response = self.unify.datasets.delete_by_resource_id("115")
+        self.assertEqual(response.status_code, 204)
+
+    @responses.activate
+    def test_attr_config_collection(self):
+        attr_config_url = self.url_prefix + "projects/1/attributeConfigurations/20"
+
+        responses.add(
+            responses.GET, self.mastering_project, json=self.mastering_project_config
+        )
+        responses.add(responses.DELETE, attr_config_url, status=204)
+
+        attr_config_collection = self.unify.projects.by_resource_id(
+            "1"
+        ).attribute_configurations()
+        response = attr_config_collection.delete_by_resource_id("20")
+        self.assertEqual(response.status_code, 204)
+
+    @responses.activate
+    def test_attributes_collection(self):
```

Comment:
BaseCollection needs to know nothing about AttributeCollection; this test needs to move out of this file and into an AttributeCollection test file